main/main.c

```diff
@@ -659,7 +659,40 @@ PHPAPI ZEND_COLD void php_log_err_with_severity(char *log_message, int syslog_ty
 	if (PG(error_log) != NULL) {
 #ifdef HAVE_SYSLOG_H
 		if (!strcmp(PG(error_log), "syslog")) {
-			php_syslog(syslog_type_int, "%s", log_message);
+			static char nvt[] = " !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~";
```

Comment:
@bukka that looks like the behavior i've seen, yeah.

@pprindeville the old RFC tells explicitly "Other code sets MAY be used as long as the characters used in the MSG are exclusively visible characters and spaces similar to those described above." In further - of course those ranges are available, but not used as a part of a code point in UTF-8. With single and double byte encodings the situation is similar. Non printable bytes don't belong to the message, that's what both RFCs tell. Combining both facts, bytes above 0x7f are fine, and bytes below 0x20 should be filtered. Especially now that we know UTF-8 is explicitly supported by the newer RFC :)

Thanks.